### PR TITLE
Change annotation names on columns

### LINKED ISSUE #97: Implement change annotation name
Implement annotation *name* changes on model objects.

Two ways how this could work:

1. User changes the qName of all annotations in a column:
  - [x] ~Direct editing of the column header cell (is this useful?)~ Impossible, as selecting the column header selects all cells in the column.
  - [x] Context menu item on the column headers
    **Open issues**
    - [x] When token column is moved somewhere else than the first column, the menu is displayed/not displayed wrongly
    - [x] Menu is displayed over table body
    - [x] When 2 editors are open, changes made in the one don't reflect in the other: listen to the event!

### PR DESCRIPTION
**Pull request by sdruskat**
## Pull request type

 

- Feature

## What is the current behavior?

Issue Number: Fixes #97 

- Supersedes #182 

## What is the new behavior?

- Adds a popup menu item "Change annotation name" on column headers
- User can change the annotation namespace and name via a popup dialog
- Annotation name is changed in the data model

## Does this introduce a breaking change?

N/A

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added **or** the PR is neither a bug fix nor a feature
- [x] Docs have been reviewed and added / updated if needed **or** the PR is neither a bug fix nor a feature
- [x] The *Unreleased* section in CHANGELOG.md has been amended to reflect the changes in this PR if needed
- [x] Build (`mvn verify`) was run locally and any changes were pushed
- [x] The pull request is against the correct branch (`master` for bug fixes, `develop` for new functionality)
- [x] [Dependencies and citation templates](https://github.com/hexatomic/hexatomic/tree/develop/releng/templates) have been updated where necessary **or** there are no new dependencies

---

# Checklist for maintainers

## Releases

- Do **NOT** push the green merge button on GitHub.  
Follow the latest version of the [developer/maintainer docs](https://hexatomic.github.io/hexatomic/dev/) for making releases.
- Check that license and citation information for dependencies are complete (each dependency has a folder in `/THIRD-PARTY/`).

### CODE REVIEW

File: docs/user/src/usage/grid-editor/editing.md [hunk 1]
```diff
@@ -42,7 +42,34 @@ As with the single cell editor, you can cancel the edit by pressing <kbd>Esc</kb
 You can also do both adding and changing at once.
 If you have selected a mixture of empty cells and ones with existing annotation values, the new value you commit in the *multi-cell editr window* will be set to all cells alike.
 
-## Deleting annotations
+# Editing annotation names in the grid
```
Comment (thomaskrause) [inline docs/user/src/usage/grid-editor/editing.md:45]: Should this be a subsection?
```suggestion
## Editing annotation names in the grid
```

File: docs/user/src/usage/grid-editor/editing.md [hunk 2]
```diff
@@ -42,7 +42,34 @@ As with the single cell editor, you can cancel the edit by pressing <kbd>Esc</kb
 You can also do both adding and changing at once.
 If you have selected a mixture of empty cells and ones with existing annotation values, the new value you commit in the *multi-cell editr window* will be set to all cells alike.
```
Comment (thomaskrause) [inline docs/user/src/usage/grid-editor/editing.md:43]: Not related to this PR directly, but I found a typo.
```suggestion
If you have selected a mixture of empty cells and ones with existing annotation values, the new value you commit in the *multi-cell editor window* will be set to all cells alike.
```

File: bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/handlers/DisplayAnnotationRenameDialogOnColumnCommandHandler.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  public DisplayAnnotationRenameDialogOnColumnCommandHandler(ColumnHeaderLayer columnHeaderLayer) {
+    this.columnHeaderLayer = columnHeaderLayer;
+  }
+
+  @Override
+  protected boolean doCommand(DisplayColumnRenameDialogCommand command) {
+    log.debug("Executing command {}.", getCommandClass().getSimpleName());
+    int columnPosition = command.getColumnPosition();
+    String originalQName = this.columnHeaderLayer.getOriginalColumnLabel(columnPosition);
+    String newQName = this.columnHeaderLayer.getRenamedColumnLabel(columnPosition);
```
Comment (thomaskrause) [inline bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/handlers/DisplayAnnotationRenameDialogOnColumnCommandHandler.java:58]: I do not understand this line. We are about the present the `AnnotationRename` dialog to the user, will this `newQName`ever be different to `originalQname` at this point?

**Review by thomaskrause — APPROVED** [review]
All in all, good to go, but I have some questions that should be clarified. It would be nice to use the `SaltUtils` class from Salt itself for generating and splitting the qualified name to avoid double implementations.

Maybe I missed some part, but as far as I understand the only way to change the annotation name is currently using the context menu, right?

File: bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/data/DataUtil.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+ * 
+ * @author Stephan Druskat (mail@sdruskat.net)
+ */
+public class DataUtil {
+
+  private DataUtil() {
+    // Should not be initiated.
+  }
+
+  /**
```
[the remark this answers is not in the document]
Comment (sdruskat) [reply bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/data/DataUtil.java:37]: Re-use is the answer.

File: bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/data/DataUtil.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+   * Builds a qualified annotation name from a namespace string and a name string.
+   * 
+   * @param namespace The namespace string
+   * @param name The name string
+   * @return The qualified annotation name
+   */
+  public static String buildQName(String namespace, String name) {
```
[the remark this answers is not in the document]
Comment (sdruskat) [reply bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/data/DataUtil.java:99]: Agree.

File: docs/user/src/usage/grid-editor/editing.md [hunk 6]
```diff
@@ -42,7 +42,34 @@ As with the single cell editor, you can cancel the edit by pressing <kbd>Esc</kb
...
+
+If the column header label includes a double colon `::`, the qualified name consists of a namespace and a name.
+
+If the label does *not* include a double colon `::`, the qualified annotation name consists only of a name.
+In this case, you can add a namespace.
+
+If the label does *end* with a double colon `::`, the qualified annotation name consists only of a namespace.
+In this case, you can add a name.
+
+## Changing multiple annotation names at once
```
[the remark this answers is not in the document]
Comment (sdruskat) [reply docs/user/src/usage/grid-editor/editing.md:59]: > I find that a little bit confusing. Why technically correct, this subsection is not only about changing _multiple_ annotation values, but about changing a single one, too. 

Not exactly, this subsection *is* about changing *multiple* annotation *names*, as there is currently no way to change the *name* of a single annotation in a single operation.

>As far as I know, the only way to change the annotation name is using the context menu. This changes both single and multiple columns.

No, you can only change the annotation for all cells in a single column at once.

[the remark this answers is not in the document]
Comment (sdruskat) [reply docs/user/src/usage/grid-editor/editing.md]: I've still edited the doc structure and the title of this section to be less confusing.

File: docs/user/src/usage/grid-editor/editing.md (hunk 2 above)
Comment (sdruskat) [reply docs/user/src/usage/grid-editor/editing.md]: Thanks for catching this, fixed.

File: bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/handlers/DisplayAnnotationRenameDialogOnColumnCommandHandler.java (hunk 3 above)
Comment (sdruskat) [reply bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/handlers/DisplayAnnotationRenameDialogOnColumnCommandHandler.java:58]: No it doesn't. That's a remnant of an older refactoring.

**Pull request merged**